customer-models/src/main/java/org/powertac/customer/evcharger/DemandSampler.java

```diff
@@ -157,6 +167,15 @@ List<DemandElement> sample (final int hod, final int popSize, final double charg
     if (!isEnabled()) {
       return new ArrayList<DemandElement>();
     }
+    
+    // We set a new seed using the randomSeedGenerator each timeslot to make
+    // sure that experiments are reproducible for the same random seed (see
+    // initialize method).
+    if (randomSeedGenerator != null) {
+      setCurrentSeed(randomSeedGenerator.nextLong(Long.MIN_VALUE,
+                                                  Long.MAX_VALUE));
+    }
```

Comment:
Calling multiple times `.nextLong` will return the same sequence of values for the same `RandomSeed` objects used in the `initialize` method.